src/main/kotlin/Issues.kt

```diff
@@ -0,0 +1,233 @@
+/*
+Issues.kt
+-----------------
+Revision History
+Rev 1 - 6/30/2024 Modified by Eitan
+        - Added Issue data type and functions operating on issues
+
+-------------------------------------------
+This file contains the abstraction of the information of an
+issue into a data type representing this information.
+Furthermore, Specific operations on issues, such as creating, editing
+and searching for issues are also present in this file.
+---------------------------------
+ */
+
+// ------
+import kotlin.time.TimeSource
+
+// ------
+
+@JvmInline
+value class Description(
+    private val description: String,
+) {
+    init {
+        require(description.length in 1..30) {
+            "Description length must be between 1 and 30 characters"
+        }
+    }
+}
+
+@JvmInline
+value class Priority(
+    private val priority: Int,
+) {
+    init {
+        require(priority in 1..5) {
+            "Priority of an issue must be between 1 and 5"
+        }
+    }
+}
+
+data class IssueInformation(
+    val description: Description,
+    val productName: Product,
+    val affectedRelease: ReleaseId,
+    val anticipatedRelease: ReleaseId? = null,
+    val priority: Priority,
+)
+
+@JvmInline
+value class IssueId(
+    private val id: Int,
+) {
+    init {
+        require(id in 1..<100) {
+            "Id must be a positive integer within [1, 99]"
+        }
+    }
+}
+
+data class Issue(
+    val id: IssueId,
+    val information: IssueInformation,
+    val createdAt: TimeSource,
```

Comment:
Is there a simpler way we can store the date without leaning on something like this? We may need to do some funky conversions between the DB and this type, and we will only be taking in strings of the form YYYY-MM-DD from the user. Maybe we can specify our own type like:

```kotlin
data class Date(
  val year: Int,
  val month: Int,
  val day: Int
)
```

Reply:
I understand your concern about the conversion of types to and from the DB, but I am confused about your comment about taking dates from the user. I thought that the only time we generate dates are when issues are created. Are there other cases where we generate a date?

Reply:
Oh that's a good point. I was thinking back to our requirements specification where I created BNFs for the dates - but I think you're right, none of our use cases actually get the date from users. I'll keep this in mind for when we update the req spec. then. I have no problem with your use of the TimeSource type then :)